javax/src/main/java/org/wildfly/naming/client/remote/EEInteroperability.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2022 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.naming.client.remote;
+
+
+import org.jboss.marshalling.ClassNameTransformer;
+import org.jboss.marshalling.MarshallingConfiguration;
+
+import static org.wildfly.naming.client.remote.RemoteNamingService.JAKARTAEE_PROTOCOL_VERSION;
+import static org.wildfly.naming.client.remote.RemoteNamingService.LATEST_VERSION;
+
+/**
+ * EE interoperability implementation for allowing Jakarta servers and clients communication with
+ * Javax EE endpoints.
+ *
+ * @author Flavia Rainone
+ * @author Richard Opalka
+ */
+class EEInteroperability {
+    // Batavia transformer sensible constant - it can start with either "javax." or "jakarta." if transformation was performed
+    private static final String VARIABLE_CONSTANT = "javax.ejb.FAKE_STRING";
+    public static final boolean JAKARTAEE_ENVIRONMENT = VARIABLE_CONSTANT.startsWith("jakarta");
+
+    /**
+     * Indicates if EE interoperable mode is enabled.
+     */
+    public static final boolean EE_INTEROPERABLE_MODE = Boolean.parseBoolean(
```

Comment:
If we would reorder JAKARTAEE_ENVIRONMENT and Boolean.parseBoolean() in the condition then we wouldn't need to read the system property (access the file system) if JavaEE environment is detected.